legend-engine-ide-lsp-extension-api/src/main/java/org/finos/legend/engine/ide/lsp/extension/reference/LegendReference.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2024 Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finos.legend.engine.ide.lsp.extension.reference;
+
+import java.util.Objects;
+import org.finos.legend.engine.ide.lsp.extension.text.LegendTextObject;
+import org.finos.legend.engine.ide.lsp.extension.text.TextLocation;
+
+/**
+ * Legend reference. This track the location where a pointer to another element exits, and the referenced element's location.
+ * This allows navigation from the given reference to the referenced location.
+ */
+public class LegendReference extends LegendTextObject
```

Comment:
We should add equals, hashCode, and toString methods for this class.

Reply:
Added on latest commit.